resource/graph_node_test.go

```diff
@@ -74,10 +80,44 @@ func TestConfiguredLifecycle(t *testing.T) {
 	test.That(t, node.NeedsReconfigure(), test.ShouldBeFalse)
 	test.That(t, node.UnresolvedDependencies(), test.ShouldBeEmpty)
 
+	rs := node.ResourceWithState()
+	test.That(t, rs.State, test.ShouldEqual, resource.NodeStateReady)
+
 	lifecycleTest(t, node, []string(nil))
 }
 
+func verifyStateTransition(
+	t *testing.T,
+	node *resource.GraphNode,
+	expectedState resource.NodeState,
+	lastState resource.WithState,
+) resource.WithState {
```

Comment:
[nit] It looks like you don't used the returned value here? Should we just return nothing from these helpers?

Reply:
[i am using them](https://github.com/viamrobotics/rdk/pull/4075/files#diff-4bd90405fd75a317db2dbde82c00b5dbec5dcef93fb63725af573afbb4603d96R130), but maybe it's confusing... my aim here is that the given `lastState` is compared against the current state, and then the current state is returned to be asserted against at the next point in the test where we expect to see a transition